hubmap/frontend/cypress/integration/hubmap_spec.js

```diff
@@ -1,10 +1,29 @@
+/* eslint no-undef: 0 */
+// TODO: Configure eslint to recognize "cy" as a global.
+// We don't click <a> tags here since hrefs redirect to cross-domain urls, which use
+// stubs (to add/remove cookies). HTTP 304 requested urls are not handled by route/request.
 describe('HuBMAP', () => {
   beforeEach(() => {
     cy.server();
     const api = 'http://localhost:8000/api';
     cy.route(`${api}/?format=json`, 'fixture:base.json');
     cy.route(`${api}/colors/?format=json`, 'fixture:colors.json');
     cy.route(`${api}/genes/?format=json`, 'fixture:genes.json');
+    // since redirect urls ( multiple HTTP 303/304) are not intercepted by route, are not recommended in cypress.
+    cy.setCookie('email', 'test@gmail.com');
+  });
+
+  after(() => {
+    cy.server()
+      .clearCookie('email');
+    const api = 'http://localhost:8000/';
+    cy.route(`${api}/api/?format=json`, 'fixture:base.json');
+    cy.route(`${api}/api/colors/?format=json`, 'fixture:colors.json');
+    cy.route(`${api}/api/genes/?format=json`, 'fixture:genes.json');
+    cy.visit('/');
```

Comment:
Again, this adds an extra unnecessary visit to '/' every test. `after` rarely needed. I think you want to check that "Login" is present when the cookie is cleared? Just make a new `it` for that:
```javascript
it('Displays "Login" when cookie is cleared',  () => {
  cy.server().clearCookie('email');
  cy.visit('/');
  cy.contains('Login');
});
```

Reply:
we should check this, whether it is a test we would like to have it in long run or not, is different. if we are testing login/logout, let us do thorough tests.

Reply:
well I have been telling i am 2 days old in cypress. And also I asked some questions on your review comments. You are not addressing my questions, in order to avoid this kind of situation. It is really good, if you can be very **specific** in your review comments, given that I , perhaps *we* have established I am new to cypress ?

Reply:
If we wanted to test code that set or cleared cookies, perhaps, but the cookie state is controlled by our tests. There is no interesting state that would persist across a new visit. We are just interested in whether the display is appropriate when the cookie is present, and when it is absent, and the later case should be a separate test.